packages/block-editor/src/components/media-upload/test/index.native.js

```diff
@@ -33,7 +35,7 @@ describe( 'MediaUpload component', () => {
 		expect( wrapper ).toBeTruthy();
 	} );
 
-	it( 'shows right media capture option for media type', () => {
+	describe( 'Media capture options for different media block types', () => {
 		const expectOptionForMediaType = ( mediaType, expectedOption ) => {
```

Comment:
As an additional note, I noticed that the second argument's name `expectedOption` is singular but in the new test cases added ([reference](https://github.com/WordPress/gutenberg/blob/d5394d0e044cb0ae25f5f9a9bcae73f2182d325c/packages/block-editor/src/components/media-upload/test/index.native.js#L60-L83)), we're passing an array. This might be also producing failures in the unit tests as the [`ByText` query](https://callstack.github.io/react-native-testing-library/docs/api-queries/#bytext) expects either a `string` or a regular expression, not an array.

Reply:
> I think part of the failures in the unit tests is related to this line, [findByText function](https://callstack.github.io/react-native-testing-library/docs/api-queries/#findby) returns a Promise so we should update expectOptionForMediaType function to be asynchronous, as well as the test cases where it's used.

Good callout on making `expectOptionForMediaType` asynchronous, and also adding an expect statement within the function too. I've added both of these changes. 

> As an additional note, I noticed that the second argument's name expectedOption is singular but in the new test cases added ([reference](https://github.com/WordPress/gutenberg/blob/d5394d0e044cb0ae25f5f9a9bcae73f2182d325c/packages/block-editor/src/components/media-upload/test/index.native.js#L60-L83)), we're passing an array. This might be also producing failures in the unit tests as the [ByText query](https://callstack.github.io/react-native-testing-library/docs/api-queries/#bytext) expects either a string or a regular expression, not an array.

I think this is a good callout as well. To be honest, I wasn't sure if passing the text strings as an array to `expectOptionForMediaType` was really improving the tests' code readability, or if it was just making things more complicated. (This is also why the PR is still a draft as I wanted to get some feedback on this approach, so thank you. 🙇 ) After thinking about it a bit more, I think keeping `getByText` as the query is better, so I used `forEach` to iterate over each option in the array:

```
await expectedOptions.forEach( ( item ) => {
    const option = wrapper.getByText( item );
    expect( option ).toBeVisible();
} );
```

What do you think about this approach @fluiddot ?

Reply:
>  think this is a good callout as well. To be honest, I wasn't sure if passing the text strings as an array to expectOptionForMediaType was really improving the tests' code readability, or if it was just making things more complicated. (This is also why the PR is still a draft as I wanted to get some feedback on this approach, so thank you. 🙇 )

To be honest, I don't have a strong opinion on whether having that function accepts one option or multiple 🤔 .

Regarding this function, my gut feeling is that the logic could be eventually moved to a generic helper, as it's likely to be used on other test cases. So, in that case, I'd advocate having the one option only so we keep the helper simple, and then let the test cases handle the multiple If I had to choose one, probably I'd go with the single option. I think it might even help with the readability of the tests.

> After thinking about it a bit more, I think keeping `getByText` as the query is better, so I used `forEach` to iterate over each option in the array:
> 
> ```
> await expectedOptions.forEach( ( item ) => {
>     const option = wrapper.getByText( item );
>     expect( option ).toBeVisible();
> } );
> ```
> 
> What do you think about this approach @fluiddot ?

I think that's a good approach 👍. The only thing I'd like to mention is that if we use `getByText`, since it's a synchronous function, we wouldn't need to use `await`, and probably make `expectOptionForMediaType` asynchronous.